kernel/src/client/sync/json.rs

```diff
@@ -11,10 +11,10 @@ use arrow_select::concat::concat_batches;
 use itertools::Itertools;
 use tracing::debug;
 
-use super::data::SimpleData;
+use crate::client::arrow_data::ArrowEngineData;
 
-pub(crate) struct SimpleJsonHandler {}
-impl JsonHandler for SimpleJsonHandler {
+pub(crate) struct SyncJsonHandler {}
```

Comment:
nit: does this work (see declaration of `SyncFileSystemClient` above)?
```suggestion
pub(crate) struct SyncJsonHandler;
```
(again below in parquet.rs)